packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -153,11 +154,17 @@ export class CdkToolkit {
       warning('⚠️ It should only be used for development - never use it for your production Stacks!');
     }
 
+    const stacks = stackCollection.stackArtifacts;
```

Comment:
This all works because `stackArtifacts` is pre-sorted based on stack dependencies